packages/cli/src/cli.ts

```diff
@@ -218,6 +243,13 @@ for (let i = 0; i < argv.length; i++) {
         flagArgConfigfile = true;
         argv.splice(i, 1);
         i--;
+    } else if (argv[i].match(/^--(http|coap|mqtt)\.port$/i)) {
+        flagArgPort = true;
+        const matches = argv[i].match(/^--(http|coap|mqtt)\.port$/i);
+        currentProtocolPort = matches[1];
+        servientPorts.set(currentProtocolPort, 0);
```

Comment:
I think it is safer to `set` the protocol only in the if branch above. `undefined` is usually recognized as "please use the default".